components/ai_chat/resources/page/state/data-context-provider.tsx

```diff
@@ -123,6 +123,9 @@ function DataContextProvider(props: DataContextProviderProps) {
   const isCharLimitExceeded = inputText.length >= MAX_INPUT_CHAR
   const isCharLimitApproaching = inputText.length >= CHAR_LIMIT_THRESHOLD
   const inputTextCharCountDisplay = `${inputText.length} / ${MAX_INPUT_CHAR}`
+  const isPremiumUserDisconnected =
+    premiumStatus === mojom.PremiumStatus.ActiveDisconnected &&
+    currentModel?.options.customModelOptions === undefined
```

Comment:
Unless `customModelOptions` can be some falsey value I think

```suggestion
    !currentModel?.options.customModelOptions
```

would be more idiomatic

Reply:
Don't we have a `isLeoModel(model)` function?